### PR TITLE
[flashlight-text] New port

### PR DESCRIPTION
**Pull request by jacobkahn**
Add a new port to Flashlight Text (https://github.com/flashlight/text). Since the Python bindings are available on PyPI, this port only adds support for building and using the C++ libraries.
- Port runs on essentially all platforms.
- Optional KenLM dependency constrains platforms/uses the KenLM port's internal platform support

-----

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/flashlight-text/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  "license": "MIT",
+  "dependencies": [
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
+    }
```
Comment (jimwang118) [inline ports/flashlight-text/vcpkg.json:15]:
```suggestion
    }
```

File: ports/flashlight-text/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,37 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO flashlight/text
+    REF v0.0.3
```
Comment (jimwang118) [inline ports/flashlight-text/portfile.cmake:4]:
```suggestion
    REF "v${VERSION}"
```

**Review by jimwang118 — CHANGES_REQUESTED** [review]

**Comment by jimwang118** [comment]
Note: I will be converting your PR to draft status. The above suggested changes are only recommendations. If you are willing to adopt them, please click "ready for review" after making the modifications. If you do not wish to make any changes, please click "ready for review" directly. That way, I can be aware that you've responded since you can't modify the tags.

File: ports/flashlight-text/portfile.cmake [hunk 3]
```diff
@@ -10,6 +10,7 @@ vcpkg_from_github(
 set(FL_DEFAULT_VCPKG_CMAKE_FLAGS
   -DFL_TEXT_BUILD_TESTS=OFF
   -DFL_TEXT_BUILD_STANDALONE=OFF
+  -DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=ON # flashlight-text doesn't explicitly export symbols
```
Comment (dg0yt) [inline ports/flashlight-text/portfile.cmake]: [Do not add CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS.](https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#do-not-add-cmake_windows_export_all_symbols)

**Review by jimwang118 — DISMISSED** [review]

File: ports/flashlight-text/portfile.cmake (hunk 3 above)
Comment (jacobkahn) [reply ports/flashlight-text/portfile.cmake]: Removing upstream -- will update PR.

Comment (jacobkahn) [reply ports/flashlight-text/portfile.cmake]: https://github.com/flashlight/text/pull/70

File: ports/flashlight-text/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+    ${FL_TEXT_DEFAULT_VCPKG_CMAKE_FLAGS}
+    ${FEATURE_OPTIONS}
+)
+vcpkg_cmake_install()
+
```
Comment (jimwang118) [inline ports/flashlight-text/portfile.cmake:28]: `vcpkg_cmake_config_fixup(CONFIG_PATH share/${PORT}/cmake)`
Add the above statement and modify the path of the `${PORT}-config` file.
When I use the following `cmake` statement to call this port, the following error occurs during compilation.
```
find_package(flashlight-text CONFIG REQUIRED)
target_link_libraries(testflashlight-text PRIVATE flashlight::flashlight-text flashlight::flashlight-text-kenlm)
```
```
Error		CMake Error at testflashlight-text/CMakeLists.txt:9 (target_link_libraries):
  Target "testflashlight-text" links to:

    flashlight::flashlight-text

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.		F:\testwork\testflashlight-text\testflashlight-text/CMakeLists.txt	9
```

File: ports/flashlight-text/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    REF v${VERSION}
+    SHA512 d9ed9b687c441e356b19a035ecef1abc76d90e3e0d8a9c32899bd7a81e379a9ae51bc6d68e7bae10cf83b87ec76289afbd81909d70799ca8d24f996e8667ad85
+    HEAD_REF main
+)
+
+# Default flags
+set(FL_TEXT_DEFAULT_VCPKG_CMAKE_FLAGS
+  -DFL_TEXT_BUILD_TESTS=OFF
+  -DFL_TEXT_BUILD_STANDALONE=OFF
+)
```
Comment (dg0yt) [inline ports/flashlight-text/portfile.cmake:13]: Move these flags directly to `vcpkg_cmake_configure`.

File: ports/flashlight-text/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+vcpkg_check_features(
+    OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+    kenlm FL_TEXT_USE_KENLM
```
Comment (dg0yt) [inline ports/flashlight-text/portfile.cmake:18]: Indent after keywords:
~~~cmake
vcpkg_check_features(
    OUT_FEATURE_OPTIONS FEATURE_OPTIONS
    FEATURES
        kenlm FL_TEXT_USE_KENLM
)

File: ports/flashlight-text/portfile.cmake [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+    ${FL_TEXT_DEFAULT_VCPKG_CMAKE_FLAGS}
+    ${FEATURE_OPTIONS}
+)
```
Comment (dg0yt) [inline ports/flashlight-text/portfile.cmake:26]: Indent after keywords:
~~~cmake:
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
    OPTIONS
        ${FL_TEXT_DEFAULT_VCPKG_CMAKE_FLAGS}
        ${FEATURE_OPTIONS}
)

File: ports/flashlight-text/portfile.cmake (hunk 4 above)
Comment (jacobkahn) [reply ports/flashlight-text/portfile.cmake:29]: The config was being installed to the incorrect location. Fix is ready for review.

File: ports/flashlight-text/portfile.cmake [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+        -DFL_TEXT_BUILD_STANDALONE=OFF
+        ${FEATURE_OPTIONS}
+    OPTIONS_DEBUG
+        "-DFL_INSTALL_CMAKE_DIR=${CURRENT_PACKAGES_DIR}/debug/share/${PORT}"
+    OPTIONS_RELEASE
+        "-DFL_INSTALL_CMAKE_DIR=${CURRENT_PACKAGES_DIR}/share/${PORT}"
+
+)
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(PACKAGE_NAME ${PORT})
```
Comment (dg0yt) [inline ports/flashlight-text/portfile.cmake:28]: `${PORT}` is the default `PACKAGE_NAME`. Omit it.

**Comment by jimwang118** [comment]
All features passed with following triplets:

```
x86-windows
x64-windows
x64-windows-static
```
Usage test pass with following triplets:

```
x86-windows
x64-windows
x64-windows-static
```

**Pull request merged**